client/keys/show.go

```diff
@@ -18,30 +22,67 @@ const (
 	FlagPublicKey = "pubkey"
 	// FlagBechPrefix defines a desired Bech32 prefix encoding for a key.
 	FlagBechPrefix = "bech"
+
+	flagMultiSigThreshold = "multisig-threshold"
 )
 
+var _ keys.Info = (keys.Info)(nil)
+
+type multiSigKey struct {
+	name string
+	key  crypto.PubKey
+}
+
+func (m multiSigKey) GetName() string            { return m.name }
+func (m multiSigKey) GetType() keys.KeyType      { return keys.TypeLocal }
+func (m multiSigKey) GetPubKey() crypto.PubKey   { return m.key }
+func (m multiSigKey) GetAddress() sdk.AccAddress { return sdk.AccAddress(m.key.Address()) }
+
 func showKeysCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "show [name]",
 		Short: "Show key info for the given name",
 		Long:  `Return public details of one local key.`,
-		Args:  cobra.ExactArgs(1),
+		Args:  cobra.MinimumNArgs(1),
 		RunE:  runShowCmd,
 	}
 
 	cmd.Flags().String(FlagBechPrefix, "acc", "The Bech32 prefix encoding for a key (acc|val|cons)")
 	cmd.Flags().Bool(FlagAddress, false, "output the address only (overrides --output)")
 	cmd.Flags().Bool(FlagPublicKey, false, "output the public key only (overrides --output)")
+	cmd.Flags().UintP(flagMultiSigThreshold, "m", 1, "K out of N required signatures")
 
 	return cmd
 }
 
-func runShowCmd(cmd *cobra.Command, args []string) error {
-	name := args[0]
+func runShowCmd(cmd *cobra.Command, args []string) (err error) {
+	var info keys.Info
 
-	info, err := GetKeyInfo(name)
-	if err != nil {
-		return err
+	if len(args) == 1 {
+		info, err = GetKeyInfo(args[0])
+		if err != nil {
+			return err
+		}
+	} else {
+		pks := make([]crypto.PubKey, len(args))
+		for i, keyName := range args {
+			info, err := GetKeyInfo(keyName)
+			if err != nil {
+				return err
+			}
+			pks[i] = info.GetPubKey()
+		}
+
+		multisigThreshold := viper.GetInt(flagMultiSigThreshold)
+		err = validateMultisigThreshold(multisigThreshold, len(args))
+		if err != nil {
+			return err
+		}
+		multikey := multisig.NewPubKeyMultisigThreshold(multisigThreshold, pks)
+		info = multiSigKey{
+			name: "multi",
```

Comment:
This should probably be a constant.

Reply:
It's a mocked value - there is no actual name for multisig keys. I can move it to a const though